File: December - 10/Solution.java
class ListNode {
      int val;
      ListNode next;
      ListNode() {}
      ListNode(int val) { this.val = val; }
      ListNode(int val, ListNode next) { this.val = val; this.next = next; }
  }
class Solution {
   
    public static ListNode reverseLinkedList(ListNode head) {
        ListNode prev = null;
        ListNode current = head;
        ListNode next = null;

        while (current != null) {
            next = current.next;
            current.next = prev;
            prev = current;
            current = next;
        }
        return prev;
    }

    public static void reorderList(ListNode head) {
        if (head == null || head.next == null) return;
        ListNode fast = head;
        ListNode slow = head;
        while (fast != null && fast.next != null) {
            fast = fast.next.next;
            slow = slow.next;
        }
        slow = reverseLinkedList(slow);
        ListNode ptr1 = head;
        ListNode ptr2 = slow;
        ListNode prev = null;
        while (ptr1 != null && ptr2 != null) {
            ListNode temp1 = ptr1.next;//2
            ListNode temp2 = ptr2.next;//5
            ptr1.next = ptr2;
            ptr2.next = temp1;
            prev = ptr2;
            ptr1 = temp1;
            ptr2 = temp2;
        }
        if (prev != null) {
            prev.next = null;
        }
    }
}

File: December - 10/Solution.py
# Enter your code here
class Node:
    def __init__(self,data):
        self.data = data
        self.next=None
class Solution:
    def Find_solution(self,head):
        if not head or not head:
            return head
            
        slow = head
        fast = head.next
        
        while fast and fast.next:
            
            fast = fast.next.next
            slow = slow.next
        
        current = slow.next
        slow.next=None
        
        prev = None
        
        #reversed
        
        while current:
            nxt = current.next
            current.next = prev
            prev = current
            current = nxt
            
        first = head
        second = prev
        
        while second:
            t1=first.next
            t2=second.next
            
            first.next=second
            second.next=t1
            
            first = t1
            second = t2
            
        return head
        
    def printList(self,head):
        if not head or not head.next:
            return head
        current = self.Find_solution(head)
        while current:
            print(current.data,end=" ")
            current = current.next
            
n = int(input("Enter N: "))

l=list(map(int,input("Enter node values: ").split()))
head = None
tail = None

for i in l:
    node = Node(i)
    if not head:
        head = tail = node
    else:
        tail.next=node
        tail=node
f2=Solution()
f2.printList(head)